lib/msal-browser/src/interaction_client/BaseInteractionClient.ts

```diff
@@ -85,168 +85,180 @@ export abstract class BaseInteractionClient {
     ): Promise<void>;
+}
 
-    protected async clearCacheOnLogout(
-        account?: AccountInfo | null
-    ): Promise<void> {
-        if (account) {
-            if (
-                AccountEntityUtils.accountInfoIsEqual(
-                    account,
-                    this.browserStorage.getActiveAccount(),
-                    false
-                )
-            ) {
-                this.logger.verbose("Setting active account to null");
-                this.browserStorage.setActiveAccount(null);
-            }
-            // Clear given account.
-            try {
-                this.browserStorage.removeAccount(
-                    AccountEntityUtils.generateAccountCacheKey(account),
-                    this.correlationId
-                );
-                this.logger.verbose(
-                    "Cleared cache items belonging to the account provided in the logout request."
-                );
-            } catch (error) {
-                this.logger.error(
-                    "Account provided in logout request was not found. Local cache unchanged."
-                );
-            }
-        } else {
-            try {
-                this.logger.verbose(
-                    "No account provided in logout request, clearing all cache items.",
-                    this.correlationId
-                );
-                // Clear all accounts and tokens
-                this.browserStorage.clear(this.correlationId);
-                // Clear any stray keys from IndexedDB
-                await this.browserCrypto.clearKeystore();
-            } catch (e) {
-                this.logger.error(
-                    "Attempted to clear all MSAL cache items and failed. Local cache unchanged."
-                );
-            }
-        }
-    }
+/**
+ *
+ * Use to get the redirect uri configured in MSAL or null.
+ * @param requestRedirectUri
+ * @returns Redirect URL
+ *
+ */
+export function getRedirectUri(
+    requestRedirectUri?: string,
+    clientConfig?: BrowserConfiguration,
+    logger?: Logger
+): string {
+    logger?.verbose("getRedirectUri called");
+    const redirectUri =
+        requestRedirectUri || clientConfig?.auth.redirectUri || "";
+    return UrlString.getAbsoluteUrl(redirectUri, BrowserUtils.getCurrentUri());
+}
 
-    /**
-     *
-     * Use to get the redirect uri configured in MSAL or null.
-     * @param requestRedirectUri
-     * @returns Redirect URL
-     *
-     */
-    getRedirectUri(requestRedirectUri?: string): string {
-        this.logger.verbose("getRedirectUri called");
-        const redirectUri = requestRedirectUri || this.config.auth.redirectUri;
-        return UrlString.getAbsoluteUrl(
-            redirectUri,
-            BrowserUtils.getCurrentUri()
-        );
-    }
+/**
+ *
+ * @param apiId
+ * @param correlationId
+ * @param forceRefresh
+ */
+export function initializeServerTelemetryManager(
+    apiId: number,
+    config: BrowserConfiguration,
```

Comment:
Do we need the whole config here or can we just pass in the clientId?

Reply:
Replaced